### DISCUSSION TITLE
SFTP provider 4.0: create_directory may not apply the expected permissions.

### DISCUSSION THREAD

**Discussion by MatthieuBlais** [apache/airflow#26226] [General] [answered]
Hi team,

We have upgraded the SFTP provider to version 4.0.0 recently, but we notice we are not able to upload files into a newly created directory using [GCSToSFTPOperator](https://airflow.apache.org/docs/apache-airflow-providers-google/6.4.0/_modules/airflow/providers/google/cloud/transfers/gcs_to_sftp.html#GCSToSFTPOperator)

We can see the directories are created with these permissions: `dr----x--t`, which we are not expecting as the [default mode](https://github.com/apache/airflow/blob/main/airflow/providers/sftp/hooks/sftp.py#L196) is 777.

After investigation, I can see there are 2 methods in the SFTP hook to create a directory: `create_directory` and `mkdir`. They use different format for passing the `mode` to paramiko:

```
# mkdir
conn.mkdir(path, mode=int(str(mode), 8))
# create_directory
conn.mkdir(path, mode=mode)
```

The first one seems to convert to octal, while the other one doesn't. Based on [paramiko doc](https://docs.paramiko.org/en/stable/api/sftp.html#paramiko.sftp_client.SFTPClient.mkdir), I believe the function expects an octal number.

If I convert 777 to octal, I get 1411, which matches with the `dr----x--t` permissions we can see.

I would think it's a bug that can be simply fixed by converting `mode` to octal in `create_directory` https://github.com/apache/airflow/compare/main...MatthieuBlais:airflow:patch-1

WDYT?

**Comment by potiuk** [comment, marked as answer]
Feel free to fix it

**Reply by MatthieuBlais** [reply]
Ok will do
